crates/primitives/src/receipt.rs

```diff
@@ -44,6 +47,79 @@ impl Receipt {
     }
 }
 
+/// A collection of receipts organized as a two-dimensional vector.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct Receipts {
+    /// A two-dimensional vector of optional `Receipt` instances.
+    pub receipt_vec: Vec<Vec<Option<Receipt>>>,
```

Comment:
given that there's only one field.. maybe we can make this a wrapper struct instead? `Receipts(Vec<Vec<Option<Receipt>>>)`

also, adding an example or reference where this is used to docs might be useful

Reply:
@joshieDo I thought about making a wrapper but I told myself that in case in the future we wanted to add other fields, it would be better to do it like that, right?

Reply:
@joshieDo As you prefer, do you want me to change?

Reply:
is fine as-is imo